apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-form.tsx

```diff
@@ -0,0 +1,546 @@
+        setIsAuthnReqSigned(initialFormValues.ISAuthnReqSigned);
+        setFormValues({ ...initialFormValues });
+    }, [ initialFormValues ]);
+
+    useEffect(() => {
+        const ifEitherOneOfThemIsChecked = isLogoutReqSigned || isAuthnReqSigned;
+        setIsAlgorithmsEnabled(ifEitherOneOfThemIsChecked);
+    }, [ isLogoutReqSigned, isAuthnReqSigned ]);
+
+    const onFormSubmit = (values: { [ key: string ]: any }): void => {
+        const manualOverride = {
+            "IncludeProtocolBinding": includeProtocolBinding,
+            "ISAuthnReqSigned": isAuthnReqSigned,
+            "IsLogoutEnabled": isLogoutEnabled,
+            "IsLogoutReqSigned": isLogoutReqSigned,
+            "IsSLORequestAccepted": isSLORequestAccepted,
+            "IsUserIdInClaims": isUserIdInClaims
+        };
+        const manualOverrideKeys = new Set<string>(Object.keys(manualOverride));
+        const authn = ({
+            ...authenticator.data,
+            properties: [
+                ...Object.keys(values)
+                    .filter((key) => !manualOverrideKeys.has(key))
+                    .map((key) => ({ key, value: values[key] })),
+                ...Object.keys(manualOverride)
+                    .map((key) => ({ key, value: manualOverride[key] })) as any
+            ]
+        });
+        onSubmit(authn);
+    };
+
+    const getAlgorithmsDropdownFieldValidators = (): (value: string) => string | undefined => {
+        if (isLogoutReqSigned || isAuthnReqSigned) {
+            return required;
+        }
+        return (/* No validations */) => void 0;
+    };
+
+    return (
+        <Form onSubmit={ onFormSubmit } uncontrolledForm={ true } initialValues={ initialFormValues }>
+
+            <Field.Input
+                required={ true }
+                name="SPEntityId"
+                value={ formValues?.SPEntityId }
+                placeholder={ t(`${ I18N_TARGET_KEY }.SPEntityId.placeholder`) }
+                ariaLabel={ t(`${ I18N_TARGET_KEY }.SPEntityId.ariaLabel`) }
+                inputType="default"
+                maxLength={ SERVICE_PROVIDER_ENTITY_ID_LENGTH.max }
+                minLength={ SERVICE_PROVIDER_ENTITY_ID_LENGTH.min }
+                label={ (
+                    <FormInputLabel htmlFor="SPEntityId">
+                        { t(`${ I18N_TARGET_KEY }.SPEntityId.label`) }
+                    </FormInputLabel>
+                ) }
+                validate={ composeValidators(
+                    required,
+                    hasLength(SERVICE_PROVIDER_ENTITY_ID_LENGTH)
+                ) }
+                hint={ (
+                    <>
+                        This value will be used as the <Code>&lt;saml2:Issuer&gt;</Code> in
+                        the SAML requests initiated from { config.ui.productName } to external
+                        Identity Provider (IdP). You need to provide a unique value
+                        as the service provider <Code>entityId</Code>.
```

Comment:
Shall we get them from localisation files? It'll be better to have that support.

Reply:
Tracking an issue for this. I will address them with https://github.com/wso2-enterprise/asgardeo-product/issues/5593